applications/nrf5340_audio/src/modules/dfu_entry.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/kernel.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include "button_assignments.h"
+#include "ble_core.h"
+#include "button_handler.h"
+#include "macros_common.h"
+
+#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
+#include <os_mgmt/os_mgmt.h>
+#endif
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include <img_mgmt/img_mgmt.h>
+#endif
+#ifdef CONFIG_MCUMGR_CMD_STAT_MGMT
+#include <stat_mgmt/stat_mgmt.h>
+#endif
+#ifdef CONFIG_MCUMGR_SMP_BT
+#include <zephyr/mgmt/mcumgr/smp_bt.h>
+#endif
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dfu, CONFIG_LOG_DFU_ENTRY_LEVEL);
+
+/* These defined name only used by DFU */
+#define DEVICE_NAME_DFU CONFIG_BT_DFU_DEVICE_NAME
+#define DEVICE_NAME_DFU_LEN (sizeof(DEVICE_NAME_DFU) - 1)
+
+/* Advertising data for SMP_SVR UUID */
+static struct bt_le_adv_param adv_param;
+static const struct bt_data ad_peer[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_UUID128_ALL,
+		      0x84, 0xaa, 0x60, 0x74, 0x52, 0x8a, 0x8b, 0x86,
+		      0xd3, 0x4c, 0xb7, 0x1d, 0x1d, 0xdc, 0x53, 0x8d),
+};
+
+/* MCUMGR related functions */
+static void mcumgr_register(void)
+{
+#if (CONFIG_AUDIO_DFU > 0)
+	/* Enable MCUMGR */
+	#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
+		os_mgmt_register_group();
+	#endif
+	#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+		img_mgmt_register_group();
+	#endif
+	#ifdef CONFIG_MCUMGR_CMD_STAT_MGMT
+		stat_mgmt_register_group();
+	#endif
+	#ifdef CONFIG_MCUMGR_SMP_BT
+		smp_bt_register();
+	#endif
+#endif
+}
+
+static void smp_adv(void)
+{
+	int ret;
+
+	ret = bt_le_adv_start(&adv_param, ad_peer, ARRAY_SIZE(ad_peer), NULL, 0);
+	if (ret) {
+		LOG_ERR("SMP_SVR Advertising failed to start (ret %d)", ret);
+		return;
+	}
+
+	LOG_INF("Regular SMP_SVR advertising started");
+}
+
+/* These callbacks are to override callback registed in module le_audio_ */
+static void dfu_connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	LOG_INF("SMP connected\n");
+}
+
+static void dfu_disconnected_cb(struct bt_conn *conn, uint8_t reason)
+{
+	LOG_INF("SMP disconnected %d\n", reason);
+	k_sleep(K_MSEC(100));
```

Comment:
Curious: why there's a 100ms sleep needed?

Reply:
Should be removed. It is accident.